ql/src/java/org/apache/hadoop/hive/ql/parse/RowResolver.java

```diff
@@ -101,17 +104,25 @@ public ASTNode getExpressionSource(ASTNode node) {
   }
 
   public void put(String tab_alias, String col_alias, ColumnInfo colInfo) {
+    if (!putInternal(tab_alias, col_alias, colInfo)) {
+      return;
+    }
+    if (col_alias != null) {
+      colInfo.setAlias(col_alias.toLowerCase());
+    }
+  }
+
+  public boolean putInternal(String tab_alias, String col_alias, ColumnInfo colInfo) {
```

Comment:
`putInternal` could be private, it's only used inside `RowResolver`

Reply:
done